Black box test for PREMIS validation events

replaceafill opened the pull request:
This converts the **PREMIS events** scenario into a scenario outline to test the **PREMIS event for validation** scenario we left out of scope in https://github.com/artefactual-labs/archivematica-acceptance-tests/pull/138 with a couple of sample transfers.

Connects to https://github.com/archivematica/Issues/issues/562

ross-spencer commented on features/steps/black_box_steps.py:
```diff
@@ -327,12 +303,44 @@ def step_impl(context, event_type):
+            files=", ".join([entry.path for entry in files_with_event_type]),
+        )
+    )
+    assert len(files_with_event_type) == expected_files_count, error
```
Nice.

ross-spencer commented:
I forgot to note in my CR as well, a play test using `behave --no-skipped -v --stop -D am_url=http://127.0.0.1:62080/ -D am_username=test -D am_api_key=test -D ss_url=http://127.0.0.1:62081/ -D ss_username=test -D ss_api_key=test --tags=black-box` worked perfectly. Nice work!

replaceafill commented on features/steps/black_box_steps.py:
```diff
@@ -9,125 +9,78 @@
+def step_impl(context, sample_transfer_path):
+    sample_transfer_path = sample_transfer_path.strip()
+    if not sample_transfer_path:
+        sample_transfer_path = "SampleTransfers/DemoTransferCSV"
```
I implemented it this way so we don't have to rewrite the other two `black-box` scenarios to contain paths in this step (granted, before grouping them, there were like 10-15 scenarios and now we only have a few). Still assuming the DemoTransfer in those seems fine to me.

ross-spencer replied:
I see. I'm not sure that will be clear to the reader, well, I'm the one struggling with that, so it might just be me! But do you think it could be summarized in an inline comment? Or perhaps as it's a constant, a module level variable with a suitable name?

replaceafill replied:
Before moving forward with this change, could you take a look at https://gist.github.com/replaceafill/1a07999505d79a1dec491e615b18f28e

Maybe making the path mandatory and explicit everywhere (including the reingest step) is better, don't you think?

replaceafill commented on features/black_box/transfer-microservices.feature:
```diff
@@ -1,22 +1,16 @@
+    Examples: sample transfers
+      | sample_transfer_path            | validated_count |
+      | SampleTransfers/DemoTransferCSV | 2               |
+      | TestTransfers/badNames          | 0               |
```
No reason for me to select `TestTransfers/badNames`. It's just that since one of our first `black-box` sprints I've kept using it to double check things. It helped me realize there's no validation events for a bunch of text files. What would be a good third option to add?

Regarding your second comment, I think we could reuse the existing [TestTransfers/acceptance-tests](https://github.com/artefactual/archivematica-sampledata/tree/master/TestTransfers/acceptance-tests) directory.

ross-spencer replied:
I see. I'm not sure the right way forward. Perhaps it's one for the sprint-wrap-up? But your rationale helps. Ideally, I think we'd have in place:

* A sample data directory specifically for automation tests.
* One of those transfer folders might be called `Transfer that doesn't generate validation events`. 

If you see where i'm going. That would be our control, and perhaps add a little more context for the reader.

replaceafill commented on features/black_box/transfer-microservices.feature:
```diff
@@ -1,22 +1,16 @@
+    And there is a message digest calculation event for each original object in the AIP METS
+    And there is a file format identification event for each original object in the AIP METS
+    And there is an ingestion event for each original object in the AIP METS
+    And there are <validated_count> original objects in the AIP METS with a validation event
```
`objects_count` seems too generic for me. For example, suppose we add PREMIS events for metadata extraction (or any other event) in AM and a step here to count how many files have that event set.

ross-spencer replied:
I'm not quite seeing the impact of that change, but what I think I'm trying to get at what the sentence is supposed to be saying. So what I think you're testing is the number of objects that have a single validation event. So then maybe to make this variable more specific and ensure it still reads okay might be `validated_objects_count`. But if we're testing something else, Like the number of validation events associated with a single object, then we might need to change the structure to something else. 

Does `validated_objects_count` work?

replaceafill replied:
:+1:

replaceafill commented:
Thank you @ross-spencer. I addressed some of your comments and left a [new question](https://github.com/artefactual-labs/archivematica-acceptance-tests/pull/142#discussion_r280922833).
